packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -10,13 +10,46 @@ import { SupportError } from './errors/index.mjs'
  * @virtual
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @type {string|undefined}
+   */
+  static moduleName
   /**
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
+   * @param {Element | null} [$module] - HTML element to use for component
    */
-  constructor() {
+  constructor($module) {
     this.checkSupport()
+    this.checkInitialised($module)
+
+    const moduleName = /** @type {typeof GOVUKFrontendComponent} */ (
+      this.constructor
+    ).moduleName
+
+    if (typeof moduleName === 'string') {
+      moduleName && $module?.setAttribute(`data-${moduleName}-init`, 'true')
+    } else {
+      throw new InitError(moduleName, this.constructor.name)
```

Comment:
**issue** `this.constructor.name` is at risk of getting mangled by Terser and other code minifiers (making `class Accordion` become `class N`) if the `name` is not defined as a static property in the class.

Reply:
hmm, would this just be a matter for adding it for our components and then adding to documentation a recommendation to specify the class name? this would be a risk for any

Reply:
@patrickpatrickpatrick I think we can do without putting the constructor's `name` inside the message. The information is already available in the stacktrace, which will have an entry for the component's constructor and let people find which component has an issue.